python/lsst/ts/m2/model.py

```diff
@@ -0,0 +1,849 @@
+                "F_A.csv",
+                "F_E.csv",
+                "F_F.csv",
+                "Tr.csv",
+                "Tu.csv",
+                "Tx.csv",
+                "Ty.csv",
+                "temp_inv.txt",
+            ]
+        )
+
+        for lut in lut_factors:
+            self.log.debug(f"Reading {lut}...")
+            name, ext = lut.split(".")
+            if ext == "csv":
+                data = pd.read_csv(
+                    config_dir / lut_path / lut,
+                    header="infer" if name.startswith("F") else None,
+                )
+                self.lut[name] = np.float64(data)
+                if name == "F_E":
+                    self.lut["lutInAngle"] = np.float64(data.keys())
+            else:
+                self.lut[name] = np.loadtxt(config_dir / lut_path / lut)
+
+    def apply_forces(self, force_axial, force_tangent):
+        """Apply force.
+
+        Parameters
+        ----------
+        force_axial : `list` or `numpy.ndarray`
+            Force of the axial actuators in Newton.
+        force_tangent : `list` or `numpy.ndarray`
+            Force of the tangent actuators in Newton.
+        """
+
+        # Check limits
+        self.check_axial_force_limit(apply=force_axial)
+        self.check_tangent_force_limit(apply=force_tangent)
+
+        self.axial_forces["applied"] = np.array(force_axial)
+        self.tangent_forces["applied"] = np.array(force_tangent)
+
+    def check_axial_force_limit(self, apply=None):
+        """Check if axial forces are out of bounds.
+
+        Parameters
+        ----------
+        apply : `numpy.ndarray` or `None`, optional
+            Forces to apply in Newton. If `None` use current setup. (default is
+            None.)
+
+        Returns
+        -------
+        `numpy.ndarray`
+            Array with the combined total force per actuator in Newton.
+
+        Raises
+        ------
+        RuntimeError
+            If force limit is out of range.
+        """
+
+        demanded_axial_force = (
+            (apply if apply is not None else self.axial_forces["applied"])
+            + self.axial_forces["lutGravity"]
+            + self.axial_forces["lutTemperature"]
+            + self.axial_forces["hardpointCorrection"]
+        )
+
+        if np.any(demanded_axial_force > self.max_axial_force):
+            actuators = np.where(demanded_axial_force > self.max_axial_force)[0]
+            raise RuntimeError(
+                f"Maximum axial force limit [{self.max_axial_force}N] reached in actuators {actuators}."
+            )
+
+        return demanded_axial_force
+
+    def check_tangent_force_limit(self, apply=None):
+        """Check if tangent forces are out of bounds.
+
+        Parameters
+        ----------
+        apply : `numpy.ndarray` or `None`, optional
+            Forces to apply in Newton. If `None` use current setup. (default is
+            None.)
+
+        Returns
+        -------
+        `numpy.ndarray`
```

Comment:
As I mentioned before, consider adding the name of the variable you are returning.

Reply:
Done in the commit: f20e3d7. Thanks!